bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -152,63 +156,71 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
         if (StringUtils.isNotBlank(proxyHost) && proxyPort != null && shouldUseProxy(url, nonProxyHosts)) {
-            client.getHostConfiguration().setProxy(proxyHost, proxyPort);
-            if (StringUtils.isNotBlank(proxyUser)) {
-                client.getState().setProxyCredentials(AuthScope.ANY,
-                        new UsernamePasswordCredentials(proxyUser, proxyPassword));
-            }
+            ProxyConfiguration proxyConfig = client.getProxyConfiguration();
+            HttpProxy proxy = new HttpProxy(proxyHost, proxyPort);
+
+            Authentication.Result result = createBasicAuthenticationResult(proxyHost, proxyPort, proxyUser,
+                    proxyPassword, true);
+            // Adding an authentication result to the authentication store allows preemptive authentication
+            client.getAuthenticationStore().addAuthenticationResult(result);
+
+            proxyConfig.getProxies().add(proxy);
         }
 
-        HttpMethod method = HttpUtil.createHttpMethod(httpMethod, url);
-        method.getParams().setSoTimeout(timeout);
-        method.getParams().setParameter(HttpMethodParams.RETRY_HANDLER, new DefaultHttpMethodRetryHandler(3, false));
+        HttpMethod method = HttpUtil.createHttpMethod(httpMethod);
+
+        Request request = client.newRequest(url).method(method).timeout(timeout, TimeUnit.MILLISECONDS);
+        RetryListener listener = new RetryListener(request, DEFAULT_NUMBER_OF_RETRIES);
+        request.onComplete(listener);
+
         if (httpHeaders != null) {
             for (String httpHeaderKey : httpHeaders.stringPropertyNames()) {
-                method.addRequestHeader(new Header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey)));
+                request.header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey));
             }
         }
         // add content if a valid method is given ...
-        if (method instanceof EntityEnclosingMethod && content != null) {
-            EntityEnclosingMethod eeMethod = (EntityEnclosingMethod) method;
-            eeMethod.setRequestEntity(new InputStreamRequestEntity(content, contentType));
+        if (method.equals(HttpMethod.POST) || method.equals(HttpMethod.PUT) && content != null) {
+            request.content(new InputStreamContentProvider(content), contentType);
         }
 
-        Credentials credentials = extractCredentials(url);
+        Map<String, String> credentials = extractCredentials(url);
         if (credentials != null) {
-            client.getParams().setAuthenticationPreemptive(true);
-            client.getState().setCredentials(AuthScope.ANY, credentials);
+            Authentication.Result result = createBasicAuthenticationResult(url, 0, credentials.get(USERNAME_KEY),
+                    credentials.get(PASSWORD_KEY), false);
+            client.getAuthenticationStore().addAuthenticationResult(result);
         }
 
         Logger logger = LoggerFactory.getLogger(HttpUtil.class);
```

Comment:
You have already introduced a static logger, so you can use LoggerFactory.getLogger in your static functions and drop the static member or you can use the static member and drop the getLogger calls.